src/dspbase.jl

```diff
@@ -186,17 +186,45 @@ conv2(A::StridedMatrix{T}, B::StridedMatrix{T}) where {T<:Integer} =
 conv2(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) where {T<:Integer} =
     round.(Int, conv2(float(u), float(v), float(A)))
 
+function check_mode_kwarg(mode::Symbol, su::Integer, sv::Integer)
+    if mode == :default_full
+        if su != sv
+            Base.depwarn(
+                "The default behavior for xcorr has changed. For more " *
+                "details, see the documentation for xcorr. To avoid this " *
+                "warning, specify mode = :full or mode = :legacy where " *
+                "appropriate",
+                :xcorr
+            )
+        end
+        :full
+    else
+        mode
+    end
+end
+
 """
-    xcorr(u,v)
+    xcorr(u,v; mode = :full)
 
-Compute the cross-correlation of two vectors.
+Compute the cross-correlation of two vectors. The size of the output depends on
+`mode` keyword argument: if `mode = :full` then the result will be the same size
+as [`conv`](@ref) of `u` and `v`. If `mode = :legacy` then the result will have
```

Comment:
maybe specify here that the length along dimension `d` will be `size(u, d)+size(v, d)-1` for clarify? I guess it could even be `length` given that this currently only supports 1D (for now...)

Reply:
Does the new docstring address this?